src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -306,53 +306,29 @@ private void validateDataTypes(List<ColumnModel> columns, Errors errors) {
     }
   }
 
-  private void validateRelationshipTerm(
-      RelationshipTermModel term,
-      List<TableModel> tables,
-      Errors errors,
-      SchemaValidationContext context) {
-    String tableName = term.getTable();
-    String column = term.getColumn();
-    if (tableName != null && column != null) {
-      if (!context.isValidTableColumn(tableName, column)) {
-        errors.rejectValue(
-            "schema",
-            "InvalidRelationshipTermTableColumn",
-            "invalid table '" + tableName + "." + column + "'");
-      }
-    }
-    for (TableModel tableModel : tables) {
-      if (term.getTable().equals(tableModel.getName())) {
-        for (ColumnModel columnModel : tableModel.getColumns()) {
-          if (term.getColumn().equals(columnModel.getName())) {
-            validateColumnType(errors, columnModel, FOREIGN_KEY);
-          }
-        }
-      }
-    }
-  }
-
   private void validateRelationship(
       RelationshipModel relationship,
       List<TableModel> tables,
       Errors errors,
       SchemaValidationContext context) {
-    RelationshipTermModel fromTerm = relationship.getFrom();
-    if (fromTerm != null) {
-      validateRelationshipTerm(fromTerm, tables, errors, context);
-    }
-
-    RelationshipTermModel toTerm = relationship.getTo();
-    if (toTerm != null) {
-      validateRelationshipTerm(toTerm, tables, errors, context);
-    }
+    ArrayList<Map<String, String>> validationErrors =
+        ValidationUtils.getRelationshipValidationErrors(relationship, tables);
+    validationErrors.forEach(e -> rejectValues(errors, e));
 
     String relationshipName = relationship.getName();
     if (relationshipName != null) {
       context.addRelationship(relationshipName);
     }
   }
 
+  private void rejectValues(Errors errors, Map<String, String> errorMap) {
+    for (var entry : errorMap.entrySet()) {
```

Comment:
I think it's totally fine but want to confirm: are we ok having the order of the error be non-deterministic?

Reply:
Good call! I changed this to use a `LinkedHashMap` instead to maintain the ordering.